packages/grafana-data/src/transformations/transformers/fillGaps.ts

```diff
@@ -0,0 +1,61 @@
+import { MutableDataFrame, DataFrame, FieldType } from '../..';
+
+import { map } from 'rxjs/operators';
+
+import { DataTransformerID } from './ids';
+import { DataTransformerInfo } from '../../types/transformations';
+
+export interface FillGapsTransformerOptions {
+  interval?: number;
+  fillWith: any;
+}
+
+export const fillGapsTransformer: DataTransformerInfo<FillGapsTransformerOptions> = {
+  id: DataTransformerID.fillGaps,
+  name: 'Fill Gaps',
+  description: '',
+  defaultOptions: {
+    fillWith: null,
+  },
+  operator: (options) => (source) =>
+    source.pipe(
+      map((dataFrames) => {
+        if (!Array.isArray(dataFrames)) {
+          return dataFrames;
+        }
+        return dataFrames.map((f) => fillGaps(f, options));
+      })
+    ),
+};
+
+const getInterval = (frame: DataFrame) => frame.fields.map((f) => f.config.interval ?? 0).find((i) => i > 0) ?? 0;
+
+export const fillGaps = (df: DataFrame, options?: FillGapsTransformerOptions): DataFrame => {
+  const frame = new MutableDataFrame(df);
+  const interval = options?.interval ?? getInterval(frame);
+  if (interval === 0) {
+    return frame;
+  }
+
+  const timeField = frame.fields.find((f) => f.type === FieldType.time);
```

Comment:
what if there are multiple time-fields? i guess for now we could just not do anything in such case?

Reply:
good question, and i'm not sure if this is the best approach. i am assuming the frame is in time series format, and i believe in the way it's used by panels the frame should already be in wide format. a similar assumption is made here: https://github.com/grafana/grafana/blob/cb53053c0e0badbef2af661c62dd6ed2d24a99f2/packages/grafana-ui/src/components/GraphNG/utils.ts#L9